### PR TITLE
Rewrite oslat run script on golang

### PR DESCRIPTION
**Pull request by cynepco3hahue**

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@fromanirh FYI you can check latest two commits.

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	oslatArgs := []string{
+		"--runtime", *runtime,
+		"--rtprio", *rtPriority,
+		"--cpu-list", updatedSelfCPUs.String(),
+		"--cpu-main-thread", mainThreadCPUSet.String(),
+	}
+
+	klog.Infof("Running the oslat command with arguments %v", oslatArgs)
+	out, err := exec.Command("/usr/bin/oslat", oslatArgs...).CombinedOutput()
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:48]: I think we should have a parameter to set the full path of `oslat`. Is fine to use this value as default.

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	cpus, err := cpuset.Parse(strings.Trim(string(out), "\n"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &cpus, nil
+}
+
+func printNodeInformation() {
+	out, err := exec.Command("cat", "/proc/cmdline").CombinedOutput()
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:74]: Why not ioutil.ReadFile?

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	if err != nil {
+		klog.Fatalf("failed to get cmdline, out: %s, err: %v", out, err)
+	}
+	klog.Infof("Environment information: /proc/cmdline: %s", string(out))
+
+	out, err = exec.Command("uname", "-nr").CombinedOutput()
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:80]: Why not https://godoc.org/golang.org/x/sys/unix#Uname ?

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"flag"
+	"os/exec"
+	"strings"
+	"time"
+
+	"k8s.io/klog"
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:9]: do we want to use klog (more deps) or the stdlib `log` package is sufficient here?

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:10]: do we want to use a simpler package (less complex dep tree) like https://github.com/fromanirh/cpuset/ ?

File: cnf-tests/testbinaries/oslat-runner/main.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	out, err := exec.Command("/usr/bin/oslat", oslatArgs...).CombinedOutput()
+	if err != nil {
+		klog.Fatalf("failed to run oslat command; out: %s; err: %v", out, err)
+	}
+
+	klog.Infof("Succeeded to run the oslat command: %s", out)
+	klog.Flush()
+}
+
+func getSelfCPUs() (*cpuset.CPUSet, error) {
```
Comment (fromanirh) [inline cnf-tests/testbinaries/oslat-runner/main.go:57]: I'm not trying to push my own https://github.com/fromanirh/numalign. I'm fine if we come out with another (set of) package(s). But we are reinventing the same wheel in few places now and we can really contribute some common package. Not in this PR, not this week, but is time to start this conversation.

**Review by fromanirh — COMMENTED** [review]
Rewriting the runner in golang seems like a good idea. We need to plan about the next steps towards shared packages for the `openshift-kni` projects. Evidence is growing there is a common set of tasks we keep reimplementing.
Not for this PR to solve.

**Comment by fromanirh** [comment]
Also, please make the commit message a bit less terse.

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 4 above)
Comment (cynepco3hahue) [reply cnf-tests/testbinaries/oslat-runner/main.go:9]: IIUC when you want to write to the file and to the stderr under the `log` you will need to set the output via ` SetOutput` to use multiple writers, it not too hard to do, but why re-invent the wheel when all deps anyway already placed under the vendor

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 1 above)
Comment (cynepco3hahue) [reply cnf-tests/testbinaries/oslat-runner/main.go:48]: will add const

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 6 above)
Comment (cynepco3hahue) [reply cnf-tests/testbinaries/oslat-runner/main.go:57]: I definitely can use it, did you think about moving it under the openshift-kni organization?

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 2 above)
Comment (cynepco3hahue) [reply cnf-tests/testbinaries/oslat-runner/main.go:74]: done

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 3 above)
Comment (cynepco3hahue) [reply cnf-tests/testbinaries/oslat-runner/main.go:80]: I will check it

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 4 above)
Comment (fromanirh) [reply cnf-tests/testbinaries/oslat-runner/main.go:9]: it's fine if we already have the deps

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 5 above)
Comment (fromanirh) [reply cnf-tests/testbinaries/oslat-runner/main.go:10]: indeed my own `cpuset` is intentionally a bare bones version, exactly to keep the baggage minimal. And if we already pull the kube deps no reason not to use the full-featured cpuset package, no problems.

File: cnf-tests/testbinaries/oslat-runner/main.go (hunk 6 above)
Comment (fromanirh) [reply cnf-tests/testbinaries/oslat-runner/main.go:57]: no need to do it now, let's just start this conversation and plan the next steps.

**Comment by cynepco3hahue** [comment]
I decided to make the change under the PR - https://github.com/openshift-kni/cnf-features-deploy/pull/317, to avoid double vendoring of the cnf-tests.

**Pull request closed without merging**